### ISSUE TITLE
Unreal game build fails - Async<bool> fails to specialize function template

### ISSUE THREAD

**Issue by borogove** [microsoft/ProjectAcoustics#7]
UE 4.23 (custom), VS2019, Wwise 2019.1.5. 

Integration instructions mostly worked as written -- a few notes:
Adding the plugin to already Wwise-integrated unreal project was a little fiddly. 
PatchWwise.bat required a checkout step, as we use Perforce. The AkAudio.Build.cs patch put the line AddWwiseLib(Target, "AcousticsWwiseFx"); after the return in GetAvailablePlatforms, which threw an error. I changed it to AddWwiseLib( "AcousticsWwiseFx" ); and moved it into the constructor at the end of a stack of other AddWwiseLib calls.
AKAudio.Build is now split up into platform-specific files so the VSVersion change was done in AkAudio_Windows.build.cs.

On build, I get several warnings and an error:
```
1>Module.ProjectAcousticsBakeUI.cpp (0:03.25 at +0:00)
1>D:/proj/phx/Dev-Main/Phoenix/Plugins/ProjectAcoustics/Source/ProjectAcousticsBakeUI/Private/AcousticsProbesTab.cpp(55): warning C4996: 'UStaticMesh::SectionInfoMap': Please do not access this member directly; use UStaticMesh::GetSectionInfoMap(). Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
1>D:\proj\phx\Dev-Main\Engine\Source\Runtime\Engine\Classes\Engine/StaticMesh.h(582): note: see declaration of 'UStaticMesh::SectionInfoMap'
1>D:/proj/phx/Dev-Main/Phoenix/Plugins/ProjectAcoustics/Source/ProjectAcousticsBakeUI/Private/AcousticsProbesTab.cpp(58): warning C4996: 'UStaticMesh::SectionInfoMap': Please do not access this member directly; use UStaticMesh::GetSectionInfoMap(). Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
1>D:\proj\phx\Dev-Main\Engine\Source\Runtime\Engine\Classes\Engine/StaticMesh.h(582): note: see declaration of 'UStaticMesh::SectionInfoMap'
1>D:/proj/phx/Dev-Main/Phoenix/Plugins/ProjectAcoustics/Source/ProjectAcousticsBakeUI/Private/AcousticsProbesTab.cpp(59): warning C4996: 'UStaticMesh::OriginalSectionInfoMap': Please do not access this member directly; use UStaticMesh::GetOriginalSectionInfoMap(). Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
1>D:\proj\phx\Dev-Main\Engine\Source\Runtime\Engine\Classes\Engine/StaticMesh.h(594): note: see declaration of 'UStaticMesh::OriginalSectionInfoMap'
1>D:/proj/phx/Dev-Main/Phoenix/Plugins/ProjectAcoustics/Source/ProjectAcousticsBakeUI/Private/AcousticsProbesTab.cpp(93): warning C4996: 'UStaticMesh::SectionInfoMap': Please do not access this member directly; use UStaticMesh::GetSectionInfoMap(). Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
1>D:\proj\phx\Dev-Main\Engine\Source\Runtime\Engine\Classes\Engine/StaticMesh.h(582): note: see declaration of 'UStaticMesh::SectionInfoMap'
1>D:/proj/phx/Dev-Main/Phoenix/Plugins/ProjectAcoustics/Source/ProjectAcousticsBakeUI/Private/AcousticsProbesTab.cpp(96): warning C4996: 'UStaticMesh::SectionInfoMap': Please do not access this member directly; use UStaticMesh::GetSectionInfoMap(). Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
1>D:\proj\phx\Dev-Main\Engine\Source\Runtime\Engine\Classes\Engine/StaticMesh.h(582): note: see declaration of 'UStaticMesh::SectionInfoMap'
1>D:/proj/phx/Dev-Main/Phoenix/Plugins/ProjectAcoustics/Source/ProjectAcousticsBakeUI/Private/AcousticsProbesTab.cpp(97): warning C4996: 'UStaticMesh::OriginalSectionInfoMap': Please do not access this member directly; use UStaticMesh::GetOriginalSectionInfoMap(). Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
1>D:\proj\phx\Dev-Main\Engine\Source\Runtime\Engine\Classes\Engine/StaticMesh.h(594): note: see declaration of 'UStaticMesh::OriginalSectionInfoMap'
1>D:/proj/phx/Dev-Main/Phoenix/Plugins/ProjectAcoustics/Source/ProjectAcousticsBakeUI/Private/AcousticsSimulationConfiguration.cpp(123): error C2672: 'Async': no matching overloaded function found
1>D:/proj/phx/Dev-Main/Phoenix/Plugins/ProjectAcoustics/Source/ProjectAcousticsBakeUI/Private/AcousticsSimulationConfiguration.cpp(133): error C2893: Failed to specialize function template 'TFuture<unknown-type> Async(EAsyncExecution,CallableType &&,TUniqueFunction<void (void)>)'
1>D:/proj/phx/Dev-Main/Phoenix/Plugins/ProjectAcoustics/Source/ProjectAcousticsBakeUI/Private/AcousticsSimulationConfiguration.cpp(133): note: With the following template arguments:
1>D:/proj/phx/Dev-Main/Phoenix/Plugins/ProjectAcoustics/Source/ProjectAcousticsBakeUI/Private/AcousticsSimulationConfiguration.cpp(133): note: 'CallableType=bool'
```

**Comment by borogove** [comment]
Ah, also, the AkAudio.Build.cs patch put the line `AddWwiseLib(Target, "AcousticsWwiseFx");` after the return in `GetAvailablePlatforms`. I changed it to `AddWwiseLib( "AcousticsWwiseFx" );` and moved it into the constructor at the end of a stack of other `AddWwiseLib` calls.

**Comment by borogove** [comment]
Looks like changing `Async<bool>( ... )` to `Async( ... )` fixes it.

**Comment by MikeChemi** [comment]
Ah, yes. Good catches. I agree with your fix. UE4.23 was released after our last update. We'll make sure to address this issue natively in our next release.

**Comment by kegodin** [comment]
This is tracked internally with bug #24293120. It's resolved as fixed internally and expected to go out with the next release.

**Comment by MikeChemi** [comment]
Fixed in version 2.0 #55
